SignalServiceKit/src/Storage/TSDatabaseView.m

```diff
@@ -341,6 +342,57 @@ + (void)asyncRegisterSecondaryDevicesDatabaseView:(OWSStorage *)storage
     [storage asyncRegisterExtension:view withName:TSSecondaryDevicesDatabaseViewExtensionName];
 }
 
++ (void)asyncRegisterLazyRestoreAttachmentsDatabaseView:(OWSStorage *)storage
+{
+    YapDatabaseViewGrouping *viewGrouping = [YapDatabaseViewGrouping withObjectBlock:^NSString *_Nullable(
+        YapDatabaseReadTransaction *transaction, NSString *collection, NSString *key, id object) {
+        if (![object isKindOfClass:[TSAttachment class]]) {
+            OWSProdLogAndFail(@"%@ Unexpected entity %@ in collection: %@", self.logTag, [object class], collection);
+            return nil;
+        }
+        if (![object isKindOfClass:[TSAttachmentStream class]]) {
+            return nil;
+        }
+        TSAttachmentStream *attachmentStream = (TSAttachmentStream *)object;
+        if (attachmentStream.lazyRestoreFragment) {
+            return TSLazyRestoreAttachmentsGroup;
+        } else {
+            return nil;
+        }
+    }];
+
+    YapDatabaseViewSorting *viewSorting = [YapDatabaseViewSorting withObjectBlock:^NSComparisonResult(
+        YapDatabaseReadTransaction *transaction,
+        NSString *group,
+        NSString *collection1,
+        NSString *key1,
+        id object1,
+        NSString *collection2,
+        NSString *key2,
+        id object2) {
+        if (![object1 isKindOfClass:[TSAttachment class]]) {
+            OWSProdLogAndFail(@"%@ Unexpected entity %@ in collection: %@", self.logTag, [object1 class], collection1);
+            return NSOrderedSame;
+        }
+        if (![object2 isKindOfClass:[TSAttachment class]]) {
+            OWSProdLogAndFail(@"%@ Unexpected entity %@ in collection: %@", self.logTag, [object2 class], collection2);
+            return NSOrderedSame;
+        }
+
+        TSAttachmentStream *attachmentStream1 = (TSAttachmentStream *)object1;
+        TSAttachmentStream *attachmentStream2 = (TSAttachmentStream *)object2;
+        return [attachmentStream2.creationTimestamp compare:attachmentStream1.creationTimestamp];
```

Comment:
At first glance this looked like a typo, but I think this is intentionally reverse chronological, right? Can you add a comment?

Reply:
Added comment.